File: frontend/src/utils/payrollPeriod.ts
import { PayrollPeriod } from "../models/employees/employees-payslips";

export const getCurrentPayrollPeriod = (): PayrollPeriod => {
  const monthIndex = new Date().getMonth();

  const periods: PayrollPeriod[] = [
    PayrollPeriod.January,
    PayrollPeriod.February,
    PayrollPeriod.March,
    PayrollPeriod.April,
    PayrollPeriod.May,
    PayrollPeriod.June,
    PayrollPeriod.July,
    PayrollPeriod.August,
    PayrollPeriod.September,
    PayrollPeriod.October,
    PayrollPeriod.November,
    PayrollPeriod.December,
  ];

  return periods[monthIndex];
};

File: backend/src/FarmsManager.Domain/Aggregates/EmployeeAggregate/Enums/PayrollPeriod.cs
using System.ComponentModel;

namespace FarmsManager.Domain.Aggregates.EmployeeAggregate.Enums;

public enum PayrollPeriod
{
    [Description("Styczeń")]
    January,

    [Description("Luty")]
    February,

    [Description("Marzec")]
    March,

    [Description("Kwiecień")]
    April,

    [Description("Maj")]
    May,

    [Description("Czerwiec")]
    June,

    [Description("Lipiec")]
    July,

    [Description("Sierpień")]
    August,

    [Description("Wrzesień")]
    September,

    [Description("Październik")]
    October,

    [Description("Listopad")]
    November,

    [Description("Grudzień")]
    December
}